src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -536,6 +547,70 @@ AppendShardIdToConstraintName(AlterTableCmd *command, uint64 shardId)
 }
 
 
+static bool
+ExtendRowReferencesWalker(Node *node, RowRefWalkerState *state)
+{
+	if (node == NULL)
+	{
+		return false;
+	}
+
+	if (IsA(node, IndexElem))
+	{
+		IndexElem *indexElem = (IndexElem *) node;
+
+		return raw_expression_tree_walker(indexElem->expr, ExtendRowReferencesWalker,
+										  state);
+	}
+	else if (IsA(node, ColumnRef))
+	{
+		ColumnRef *columnRef = (ColumnRef *) node;
+		ListCell *fieldsCell;
+
+		/*
+		 * Append the shardId to any ColumnRef String values that are
+		 * equal to the relationName.  These are actually ROW(relname)
+		 * references.
+		 */
+		foreach(fieldsCell, columnRef->fields)
```

Comment:
Here's the issue with this.

If you write the statement like this (note `wikipedia_changes.*`):

```sql
create index idxwc on wikipedia_changes using zombodb(zdb(tableoid, ctid), zdb(wikipedia_changes.*)) with (url='http://localhost:9200/');
```

Then the `ColumnRef` node looks like: `{COLUMNREF :fields ("wikipedia_changes" {A_STAR}) :location 79}`, which is what you'd expect.

But if you write the statement without the `.*`:

```sql
create index idxwc on wikipedia_changes using zombodb(zdb(tableoid, ctid), zdb(wikipedia_changes)) with (url='http://localhost:9200/');
```

The the `ColumnRef` is:  `{COLUMNREF :fields ("wikipedia_changes") :location 79}`

Which isn't really distinguishable from the other ColumnRefs in the query:  `{COLUMNREF :fields ("tableoid") :location 58}` and `{COLUMNREF :fields ("ctid") :location 68}`.

What are your thoughts here?